pkg/didcomm/protocol/didexchange/service_test.go

```diff
@@ -775,13 +780,32 @@ func handleConnectionResponseEvents(t *testing.T, svc *Service, e dispatcher.DID
 		err = svc.store.Put(id, jsonDoc)
 		require.NoError(t, err)
 
-		e.Callback = func(didCommCallback dispatcher.DIDCommCallback) {
+		e.Callback = func(didCommCallback *dispatcher.DIDCommCallback) {
 			svc.processCallback(id, didCommCallback)
 		}
 	}
 
 	// invoke callback
-	e.Callback(dispatcher.DIDCommCallback{Err: nil})
+	msg := dispatcher.NewDIDCommCallback()
+	if invalid {
+		done := make(chan struct{})
+		result, err := msg.WithResult().Result()
```

Comment:
I would suggest that the extra WithResult() is unnecessary from a caller's Point of View.

Reply:
Done